plaso/parsers/powershell_transcript.py

```diff
@@ -0,0 +1,343 @@
+    remoting_protocol_version (str): PowerShell
+      remote management protocol version.
+    runas_user (str): User context of execution.
+    serialization_version (str): Serialization method version.
+    username (str): User that executed command.
+    version (str): PowerShell version.
+    ws_man_stack_version (str): WS-Management stack version
+    username (str): User that executed command.
+  """
+
+  DATA_TYPE = 'powershell:transcript:event'
+
+  def __init__(self):
+    """Initializes event data."""
+
+    super(PowerShellTranscriptEventData,
+      self).__init__(data_type=self.DATA_TYPE)
+    self.build_version = None
+    self.clr_version = None
+    self.command = None
+    self.compatible_versions = None
+    self.config_name = None
+    self.edition = None
+    self.host_application = None
+    self.machine = None
+    self.process_id = None
+    self.remoting_protocol_version = None
+    self.runas_user = None
+    self.serialization_version = None
+    self.username = None
+    self.version = None
+    self.ws_man_stack_version = None
+
+class PowerShellTranscriptParser(text_parser.PyparsingMultiLineTextParser):
+  """Parses events from PowerShell transcript files."""
+
+  NAME = 'powershell_transcript'
+
+  DATA_FORMAT = 'PowerShell transcript event'
+
+  # PowerShell transcript lines can be very long.
+  MAX_LINE_LENGTH = 65536
+
+  _SEPARATOR = '**********************'
+
+  _COLON = pyparsing.Literal(': ').suppress()
+
+  # key always start with an uppercase character
+  _METADATA_KEY = (pyparsing.Word(pyparsing.alphas.upper(), ' ' +
+    pyparsing.alphas) + _COLON)
+  # negative lookahead needed for correct identification
+  _METADATA_VALUE = (~_METADATA_KEY +
+    pyparsing.Word(pyparsing.printables, ' ' + pyparsing.printables))
+
+  _METADATA_LINE = (
+    _METADATA_KEY +
+    _COLON +
+    pyparsing.Optional(_METADATA_VALUE) +
+    pyparsing.LineEnd().suppress()
+  )
+
+  _TRANSCRIPT_LINE = (
+    ~_METADATA_LINE +
+    pyparsing.Word(' ' + '\t' + '\r' +
+      pyparsing.printables + pyparsing.alphas8bit +
+      pyparsing.punc8bit, exclude_chars='**********************') +
+    pyparsing.LineEnd().suppress()
+  )
+
+  _SEPARATOR_LINE = (
+    pyparsing.Literal(_SEPARATOR)
+    + pyparsing.LineEnd().suppress()
+  )
+
+  LINE_STRUCTURES = [
+      ('metadata_line', _METADATA_LINE),
+      ('separator_line', _SEPARATOR_LINE),
+      ('transcript_line', _TRANSCRIPT_LINE),
+  ]
+
+  # variables needed for further processing
+  is_first_line = True
+  found_first_separator = False
+  key_structure = []
+  base_event = None
+  parser_mediator = None
+
+  def _GetTimestampFromString(self, value):
+    """Parse a timestamp string an return a TimeElements event.
+
+    Args:
+      value (str): String containing a timestamp.
+
+    Returns:
+      dfdatetime_time_elements.TimeElements: TimeElements, if string
+        was valid, None otherwise
+    """
+    if len(value) == 14:
+      time_elements_tuple = int(value[:4]),\
```

Comment:
style nit - use parenthesis around these instead of backslash line continuation and use appropriate indentation.